devito/types/grid.py

```diff
@@ -786,10 +746,22 @@ def __subdomain_finalize__(self, grid, counter=0, **kwargs):
                 sd_func.data[:, idx] = self._local_bounds[idx]
 
             dname = '%si%d' % (d.name, counter)
-            dimensions.append(MultiSubDimension(dname, d,
-                                                functions=sd_func,
-                                                bounds_indices=(2*i, 2*i+1),
-                                                implicit_dimension=i_dim))
+
+            ltkn = Symbol(name="%s_ltkn%d" % (d.name, counter), dtype=np.int32,
+                          is_const=True, nonnegative=True)
+            rtkn = Symbol(name="%s_rtkn%d" % (d.name, counter), dtype=np.int32,
+                          is_const=True, nonnegative=True)
+
+            left = d.symbolic_min + ltkn
```

Comment:
Why isn't this done in the Dimension's init?

Reply:
Seconded

Reply:
Not in `AbstractSubDimension.__init__` though, because `SubDimension` gets custom left/right depending on whether it's a left/right/middle SubDimension